### PR TITLE
fix(manager/kustomize): Don't crash on non-string resources

### PR DESCRIPTION
**Pull request by codablock**
## Changes

This PR fixes an exception that happens when a `kustomization.yaml` contains `resources` that are not strings. These might be `null` resources or any other invalid resources.

## Context

Even though non-string resources are forbidden in Kustomize, it might still happen when some form of pre-processing or templating is applied to `kustomization.yaml` files. In that case, renovate should error out when extracting packages.

This was not a problem before https://github.com/renovatebot/renovate/pull/20116 got merged as Renovate simply bailed out in `extractResource` due to non-strings not matching the regex.

The stack trace of the original exception was:
```
TypeError: base.includes is not a function
    at extractResource (/opt/buildpack/tools/renovate/34.128.3/node_modules/renovate/lib/modules/manager/kustomize/extract.ts:35:12)
    at Object.extractPackageFile (/opt/buildpack/tools/renovate/34.128.3/node_modules/renovate/lib/modules/manager/kustomize/extract.ts:192:17)
    at extractPackageFile (/opt/buildpack/tools/renovate/34.128.3/node_modules/renovate/lib/modules/manager/index.ts:70:9)
    at getManagerPackageFiles (/opt/buildpack/tools/renovate/34.128.3/node_modules/renovate/lib/workers/repository/extract/manager-files.ts:52:43)
    at /opt/buildpack/tools/renovate/34.128.3/node_modules/renovate/lib/workers/repository/extract/index.ts:57:28
    at async Promise.all (index 3)
    at extractAllDependencies (/opt/buildpack/tools/renovate/34.128.3/node_modules/renovate/lib/workers/repository/extract/index.ts:55:26)
    at extract (/opt/buildpack/tools/renovate/34.128.3/node_modules/renovate/lib/workers/repository/process/extract-update.ts:139:28)
    at extractDependencies (/opt/buildpack/tools/renovate/34.128.3/node_modules/renovate/lib/workers/repository/process/index.ts:120:26)
    at Object.renovateRepository (/opt/buildpack/tools/renovate/34.128.3/node_modules/renovate/lib/workers/repository/index.ts:56:9)
    at attributes.repository (/opt/buildpack/tools/renovate/34.128.3/node_modules/renovate/lib/workers/global/index.ts:181:11)
    at start (/opt/buildpack/tools/renovate/34.128.3/node_modules/renovate/lib/workers/global/index.ts:166:7)
    at /opt/buildpack/tools/renovate/34.128.3/node_modules/renovate/lib/renovate.ts:18:22
```
- ref #20055

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/kustomize/extract.ts [hunk 1]
```diff
@@ -32,7 +32,9 @@ const gitUrlWithPath = regEx(
 export function extractResource(base: string): PackageDependency | null {
   let match: RegExpExecArray | null;
 
-  if (base.includes('_git')) {
+  if (typeof base !== 'string') {
```
Comment (viceice) [inline lib/modules/manager/kustomize/extract.ts:35]:
```suggestion
  if (!is.string(base)) {
```

**Review by viceice — COMMENTED** [review]
needs a unit test

**Review by rarkins — APPROVED** [review]

**Review by viceice — CHANGES_REQUESTED** [review]
still needs a unit test for code coverage

**Comment by viceice** [comment]
related to
- #20055

**Comment by codablock** [comment]
Can someone else take over at this point and provide a unit test? I'm absolutely not fluent in JavaScript/TypeScript/Node and would be completely lost.

**Comment by rarkins** [comment]
Will #20338 achieve the same thing?

**Comment by codablock** [comment]
@rarkins I would assume yes, same effect. You can close this PR if you want :)

**Pull request closed without merging**